src/line_launcher/mod.rs

```diff
@@ -141,6 +138,173 @@ pub fn get_combined_message_receiver(
     receiver
 }
 
+#[derive(Clone, PartialEq, Eq, Debug, Copy)]
+struct MeasureBeat {
+    beat_number: BeatNumber,
+    measure: u32,
+}
+
+impl MeasureBeat {
+    pub fn new(beat_number: BeatNumber, measure: u32) -> Self {
+        Self {
+            beat_number,
+            measure,
+        }
+    }
+
+    pub fn increment(&mut self) {
+        if self.beat_number.sixteenth_note == 15 {
+            self.measure += 1;
+        }
+        self.beat_number.increment();
+    }
+
+    pub fn incremented(&self) -> Self {
+        let mut cloned = *self;
+        cloned.increment();
+        cloned
+    }
+
+    pub fn incremented_by(&self, num_sixteenths: u32) -> Self {
+        let mut cloned = *self;
+        for _ in 0..num_sixteenths {
+            cloned.increment();
+        }
+        cloned
+    }
+}
+
+impl Default for MeasureBeat {
+    fn default() -> Self {
+        Self {
+            beat_number: BeatNumber::new(0),
```

Comment:
In other number-ish type I've seen, it's pretty common to provide a `zero` constructor.